### PR TITLE
chore: updating CLI help screen

### PR DESCRIPTION
**Pull request by SebOpsahl**
Updating CLI help screen to show arguments for commands, and updating the help screen to show more options and commands

Changelog: None

Ticket: MEN-6986

### CODE REVIEW

**Review by lluiscampos — CHANGES_REQUESTED** [review]

**Comment by SebOpsahl** [comment]
Oh okey, fixed it all now in the latest commit :+1:

File: src/client_shared/conf/conf_cli_help.cpp [hunk 1]
```diff
@@ -147,6 +147,19 @@ void PrintInTwoColumns(
 	}
 }
 
+string PrintArgument(CliArgument argument, ostream &stream) {
+	string cliarg;
+
+	if (!argument.name.empty()) {
```
Comment (lluiscampos) [inline src/client_shared/conf/conf_cli_help.cpp:153]: This works just fine.

A semantically better approach could be to use `optional` template, where in `CliCommand` we would instead have a `optional <CliArgument> argument;` attribute and in this line we would check with `argument.has_value()`. You can read more about this template [in here](https://github.com/mendersoftware/mender/blob/e8337ce91d9f6afec81fe7e62aa8a698c55726fa/src/common/optional.hpp) and search in our codebase for examples of usage.

File: src/mender-update/cli/cli.cpp [hunk 2]
```diff
@@ -49,6 +49,11 @@ const conf::CliCommand cmd_daemon {
 const conf::CliCommand cmd_install {
 	.name = "install",
 	.description = "Mender Artifact to install - local file or a URL",
+	.argument =
+		{
+			.name = "manifest",
+			.mandatory = false,
```
Comment (lluiscampos) [inline src/mender-update/cli/cli.cpp:55]: As I indirectly wrote [here](https://github.com/mendersoftware/mender/pull/1601#discussion_r1535637076) and directly wrote [here](https://github.com/mendersoftware/mender/pull/1601#discussion_r1535638579), for `mender-update` the argument is "artifact" and it is mandatory.

**Review by lluiscampos — CHANGES_REQUESTED** [review]
Well done. The design is now fine, and I have a suggestion below for how to make it even better.

Requesting changes for my second comment (artifact instead of manifest), the first comment is a suggestion.

**Comment by lluiscampos** [comment]
> Changelog: updating CLI help screen to show arguments for commands, and updating the help screen to show more options and commands
> 
> Ticket: MEN-6986

This is not a feature and have no effects for the user. Please mark it as `chore`

File: src/client_shared/conf/conf_cli_help.cpp (hunk 1 above)
Comment (SebOpsahl) [reply src/client_shared/conf/conf_cli_help.cpp]: This sounds good, changed it to use this approach for both PR’s now 🙂

File: src/mender-update/cli/cli.cpp (hunk 2 above)
Comment (SebOpsahl) [reply src/mender-update/cli/cli.cpp]: Oh sorry, fixed it now

**Review by lluiscampos — APPROVED** [review]

**Comment by SebOpsahl** [comment]
@mender-test-bot start pipeline

**Pull request merged**